server/controllers/productController.js

```diff
@@ -0,0 +1,85 @@
+import Product from "../models/productModel.js";
+
+export const createPost = async (req, res) => {
+  try {
+    const newProduct = await Product.create(req.body);
+    res.status(201).json(newProduct);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+};
+
+export const deletedProduct = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const product = await Product.findByIdAndDelete(id);
+    if (!product) {
+      return res.status(404).send({ message: "Product not found" });
+    }
+    res.status(204).json();
+  } catch (error) {
+    res
+      .status(500)
+      .send({ message: "Error deleting item", error: error.message });
+  }
+};
+
+export const getProduct = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const product = await Product.findById(id);
+    if (!product) {
+      return res.status(404).json({ message: "Product not found" });
+    }
+    res.status(200).json(product);
+  } catch (error) {
+    console.error("Error fetching menu item:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+};
+
+export const getAllProduct = async (req, res) => {
+  const New = req.query.new;
+  const Category = req.query.category;
+  try {
+    let product;
+    if (New) {
+      product = await Product.find().sort({ createdAt: -1 }).limit(2);
```

Comment:
incorporate pagination instead of hardcoding limit value

Reply:
done.. needs review